sencha-workspace/packages/slate-cbl/src/view/teacher/demonstration/EditWindowController.js

```diff
@@ -175,6 +180,29 @@ Ext.define('Slate.cbl.view.teacher.demonstration.EditWindowController', {
             competenciesStore.on('load', _restoreSavedSkills, me, { single: true });
         }
     },
+    
+    onDeleteDemonstrationConfirm: function() {
+        var me = this,
+            editWindow = me.getView(),
+            params = Ext.Object.fromQueryString(window.location.href.split('?')[1]),
```

Comment:
This doesn't seem safe. We're relying on something to be in the URL that may not be.